### PR TITLE
Issue #984 known imod5 issues

### LINKED ISSUE #984: Expand Known Issues section
We have a known issues section, but it can be expanded.
https://deltares.github.io/imod-python/faq/known-issues.html

Known Issues to add:

- Using both the BarycentricInterpolator and OverlapRegridder when regridding causes non-alignment inactive cells at edges
- This means downscaling is now not done with the interpolator.
- FUTURE: iMOD5 backwards compatibility issues. We are not going to fix every problem in an iMOD5 database. For example I ran into the issue where the IBOUND array contained values like 0.249, 0.582, and 0.769. We do not fix this.

### PR DESCRIPTION
**Pull request by JoerivanEngelen**
Fixes #984 

# Description
- Adds "Known Issues" section to iMOD5 Backwards Compatibility page
- "Known Issues" page refers to iMOD5 Backwards compatibility page
- Set max support Python version to 3.12

### CODE REVIEW

File: docs/installation.rst [hunk 1]
```diff
@@ -4,7 +4,7 @@ Installing
 Which Python?
 -------------
 
-You'll need **Python 3.10 or greater**. 
+You'll need **Python 3.10 up until 3.12**.
```
Comment (Manangka) [inline docs/installation.rst:7]: What about rephrasing it to one of the following: 
- "We support Python 3.10 to 3.12". 
- "The supported python versions are 3.10-3.12"
- "Imod-python has been tested on Python 3.10 to 3.12"
- "Imod-Python has been developed using Python 3.10 to 3.12"
-

File: docs/faq/imod5_backwards_compatibility.rst [hunk 2]
```diff
@@ -2,9 +2,43 @@ iMOD5 backwards compatibility
...
+
+Known issues
+------------
+
+- Constants are only supported for the CAP package. For other packages, the
+  constants will cause an error.
+- ISG files cannot be read directly. The workaround is to rasterize the ISG
+  files to IDF files using the iMOD5 BATCH function ISGGRID.
+- MetaSWAP sprinkling wells defined as IPF files are not supported.
+- MetaSWAP's "flexible drainage" ("Peilgestuurde drainage" in Dutch) not
```
Comment (Manangka) [inline docs/faq/imod5_backwards_compatibility.rst:18]: not -> is not

File: docs/faq/imod5_backwards_compatibility.rst [hunk 3]
```diff
@@ -2,9 +2,43 @@ iMOD5 backwards compatibility
...
+  supported.
+- IPEST is not supported.
+
+Notes
+-----
+
+- if STO package in projectfile is absent, error is thrown when trying to write
```
Comment (Manangka) [inline docs/faq/imod5_backwards_compatibility.rst:25]: error -> an error

**Review by Manangka — APPROVED** [review]

File: docs/installation.rst (hunk 1 above)
Comment (JoerivanEngelen) [reply docs/installation.rst:7]: Changed to most explicit as possible: "You'll need **Python 3.10, 3.11, or 3.12**. "

**Pull request merged**